src/storage/src/hummock/compactor/compaction_utils.rs

```diff
@@ -228,34 +228,20 @@ pub async fn generate_splits(
     Ok(vec![])
 }
 
-pub fn estimate_task_memory_capacity(
-    context: Arc<CompactorContext>,
-    task: &CompactTask,
-) -> (usize, usize, usize) {
+pub fn estimate_task_output_capacity(context: Arc<CompactorContext>, task: &CompactTask) -> usize {
     let max_target_file_size = context.storage_opts.sstable_size_mb as usize * (1 << 20);
     let total_input_file_size = task
         .input_ssts
         .iter()
         .flat_map(|level| level.table_infos.iter())
         .map(|table| table.file_size)
         .sum::<u64>();
-    let total_input_uncompressed_file_size = task
-        .input_ssts
-        .iter()
-        .flat_map(|level| level.table_infos.iter())
-        .map(|table| table.uncompressed_file_size)
-        .sum::<u64>();
 
     let capacity = std::cmp::min(task.target_file_size as usize, max_target_file_size);
     let total_file_size = (total_input_file_size as f64 * 1.4).round() as usize;
 
-    let c = match task.compression_algorithm {
+    match task.compression_algorithm {
         0 => std::cmp::min(capacity, total_file_size),
         _ => capacity,
```

Comment:
Why _ => doesn't need `min`?

Reply:
good catch 

Even now, after we only use streaming_uploader, this estimate is not so important.

I'd still like to discuss whether we should use total_input_uncompressed_file_size for estimation? Here _=> the reason for not using min, is that in most cases, the uncompressed_file_size > capacity (compression_algorithm > 0)